pkg/lib/bundle/generate.go

```diff
@@ -46,30 +50,25 @@ type AnnotationMetadata struct {
 // @channels: The list of channels that bundle image belongs to
 // @channelDefault: The default channel for the bundle image
 // @overwrite: Boolean flag to enable overwriting annotations.yaml locally if existed
-func GenerateFunc(directory, outputDir, packageName, channels, channelDefault string, overwrite bool) error {
-	// clean the input so that we know the absolute paths of input directories
-	directory, err := filepath.Abs(directory)
-	if err != nil {
-		return err
-	}
-	if outputDir != "" {
-		outputDir, err = filepath.Abs(outputDir)
-		if err != nil {
-			return err
-		}
-	}
-
-	_, err = os.Stat(directory)
-	if os.IsNotExist(err) {
+func GenerateFunc(option ...BundleOption) error {
+	bundleConfig := &BundleConfig{}
+	bundleConfig.apply(option)
+	if err := bundleConfig.complete(); err != nil {
 		return err
 	}
 
 	// Determine mediaType
-	mediaType, err := GetMediaType(directory)
+	mediaType, err := GetMediaType(bundleConfig.bundleDir)
 	if err != nil {
 		return err
 	}
 
+	if mediaType == RegistryV1Type {
```

Comment:
I thought we would generate a CSV if it is a `plain` type meaning it doesn't have a CSV already. If it is a `registryv1` then it already has a CSV.

Reply:
oh, i should have thought about that before i add a way to detect non-inferable file